dev/com.ibm.ws.beanvalidation.v20.cdi/src/com/ibm/ws/beanvalidation/v20/cdi/internal/LibertyHibernateValidatorExtension.java

```diff
@@ -126,12 +116,38 @@ private ValidationExtension delegate(String newClassloaderHint) {
         return extDelegate;
     }
 
-    public void beforeBeanDiscovery(@Observes BeforeBeanDiscovery beforeBeanDiscoveryEvent,
-                                    final BeanManager beanManager) {
-        delegate(null).beforeBeanDiscovery(beforeBeanDiscoveryEvent, beanManager);
+    public void beforeBeanDiscovery(@Observes BeforeBeanDiscovery beforeBeanDiscoveryEvent, final BeanManager beanManager) {
+        if (!delegateFailed)
+            try {
+                delegate(null).beforeBeanDiscovery(beforeBeanDiscoveryEvent, beanManager);
+            } catch (Exception e) {
+                delegateFailed = true;
+                // TODO: Once https://github.com/OpenLiberty/open-liberty/issues/1746 is implemented, we
+                // can use the proper bean ID in the warning message instead of the BeanManager toString()
+                if (tc.isWarningEnabled())
+                    Tr.warning(tc, "UNABLE_TO_REGISTER_WITH_CDI", beanManager.toString(), e.getMessage());
```

Comment:
Would it be valuable to include the exception class along with the message?
Could switch to:
Tr.warning(tc, "UNABLE_TO_REGISTER_WITH_CDI", beanManager.toString(), e);